### PR TITLE
feat: add "uri" for MsgModify and deprecate the old ones

### PR DESCRIPTION
**Pull request by 0Tech**
## Description

closes: #899

## Checklist:

- [x] I followed the [contributing guidelines](https://github.com/line/lbm-sdk/blob/main/CONTRIBUTING.md) and [code of conduct](https://github.com/line/lbm-sdk/blob/main/CODE_OF_CONDUCT.md).
- [x] I have added a relevant changelog to `CHANGELOG.md`
- [x] I have added tests to cover my changes.
- [ ] I have updated the documentation accordingly.
- [x] I have updated API documentation `client/docs/swagger-ui/swagger.yaml`

### CODE REVIEW

File: CHANGELOG.md [hunk 1]
```diff
@@ -51,6 +51,7 @@ Ref: https://keepachangelog.com/en/1.0.0/
 * (client) [\#890](https://github.com/line/lbm-sdk/pull/890) Map Ostracon:ErrTxInMap to lbm-sdk:ErrTxInMempoolCache
 * (x/collection) [\#894](https://github.com/line/lbm-sdk/pull/894) Change the default params of x/collection
 * (ante) [\#895](https://github.com/line/lbm-sdk/pull/895) Remove max gas validation
+* (ante) [\#900](https://github.com/line/lbm-sdk/pull/900) Add uri for MsgModify and deprecate the old ones
```
Comment (0Tech) [inline CHANGELOG.md:54]:
```suggestion
* (x/collection,token) [\#900](https://github.com/line/lbm-sdk/pull/900) Add uri for MsgModify and deprecate the old ones
```

**Comment by 0Tech** [comment]
I will revise the relevant event.
as-is: emit uri or img_uri
to-be: emit both of uri and img_uri

File: go.mod [hunk 2]
```diff
@@ -102,7 +102,7 @@ require (
 	github.com/minio/highwayhash v1.0.2 // indirect
 	github.com/mitchellh/mapstructure v1.5.0 // indirect
 	github.com/mtibben/percent v0.2.1 // indirect
-	github.com/onsi/gomega v1.20.0 // indirect
+	github.com/onsi/gomega v1.22.1 // indirect
```
Comment (zemyblue) [inline go.mod:105]: Why is this changed? Do you know?

**Comment by zemyblue** [comment]
> 

Please check if no problem with related team, @0Tech .

File: go.mod (hunk 2 above)
Comment (0Tech) [reply go.mod:105]: Because our proto script runs `go mod tidy`.

File: x/token/validation.go [hunk 3]
```diff
@@ -97,3 +98,13 @@ func validateChange(change Attribute) error {
 func ValidateContractID(id string) error {
 	return class.ValidateID(id)
 }
+
+func canonicalKey(key string) string {
+	convert := map[string]string{
+		AttributeKeyImageURI.String(): AttributeKeyURI.String(),
+	}
```
Comment (dudong2) [inline x/token/validation.go:105]: AttributeKeyImageURI and AttributeKeyURI are all fixed values, so convert also does not change. It is unnecessary to declare this every time. How about pre-declaring it with the var at the top?

Comment (0Tech) [reply x/token/validation.go:105]: We should avoid using globals if it's not performance-critical. Also, there is no other place using this map.
Could you elaborate your concern on it?

**Comment by 0Tech** [comment]
> > 
> 
> Please check if no problem with related team, @0Tech .

Thank you.

For other reviewers: this PR is now under review by our clients.

**Comment by 0Tech** [comment]
> > > 
> > 
> > 
> > Please check if no problem with related team, @0Tech .
> 
> Thank you.
> 
> For other reviewers: this PR is now under review by our clients.

Done.

File: proto/lbm/collection/v1/tx.proto [hunk 4]
```diff
@@ -466,7 +466,7 @@ message MsgModify {
   // Note: if token type is of FTs, the index cannot be empty.
   string token_index = 4;
   // changes to apply.
-  // possible attribute keys on modifying collection: name, base_img_uri, meta.
+  // possible attribute keys on modifying collection: name, uri, base_img_uri (deprecated), meta.
```
Comment (zemyblue) [inline proto/lbm/collection/v1/tx.proto:469]: I think we don't need `base_image_uri` for now. Isn't it?

File: proto/lbm/token/v1/tx.proto [hunk 5]
```diff
@@ -274,7 +274,7 @@ message MsgModify {
   // the address of the grantee which must have modify permission.
   string owner = 2;
   // changes to apply.
-  // possible attribute keys are: name, img_uri, meta
+  // possible attribute keys are: name, uri, img_uri (deprecated), meta
```
Comment (zemyblue) [inline proto/lbm/token/v1/tx.proto:277]: Can the `img_url` key be used after this PR merge?

File: proto/lbm/collection/v1/tx.proto (hunk 4 above)
Comment (0Tech) [reply proto/lbm/collection/v1/tx.proto:469]: The client didn't mention about tx. So I have no idea whether we can remove it.

File: proto/lbm/token/v1/tx.proto (hunk 5 above)
Comment (0Tech) [reply proto/lbm/token/v1/tx.proto:277]: In the current implementation, yes.

**Review by zemyblue — APPROVED** [review]

**Review by jaeseung-bae — APPROVED** [review]
LGTM

File: x/collection/msgs.go [hunk 6]
```diff
@@ -221,6 +222,16 @@ func validateChange(change Attribute, validators map[string]func(string) error)
 	return validator(change.Value)
 }
 
+func canonicalKey(key string) string {
+	convert := map[string]string{
```
Comment (loin3) [inline x/collection/msgs.go:226]: Is the use of a map rather than a simple if statement to make it easier to respond when another change occurs later? For example if `Meta` is deprecated and `Metadata` should be used then just add `AttributeKeyMeta.String(): AttributeMetadata.String(),` ?

Comment (0Tech) [reply x/collection/msgs.go:226]: Exactly.

**Review by loin3 — APPROVED** [review]
